src/main/java/asg/asts/Generator.java

```diff
@@ -1115,16 +1115,16 @@ private void createToString(ListDef l, StringBuilder sb) {
         }
 
         sb.append("    @Override public String toString() {\n");
-        sb.append("        String result =  \"" + l.getName() + "(\";\n");
+        sb.append("        StringBuilder result = new StringBuilder(\"" + l.getName() + "(\");\n");
         sb.append("        boolean first = true;\n");
         sb.append("        for (" + printType(l.itemType) + " i : this ) {\n");
-        sb.append("            if (!first) { result +=\", \"; }\n");
-        sb.append("            if (result.length() > 1000) { result +=\"...\"; break; }\n");
-        sb.append("            result += i;\n");
+        sb.append("            if (!first) { result.append(\", \"); }\n");
+        sb.append("            if (result.length() > 1000) { result.append(\"...\"); break; }\n");
+        sb.append("            result.append(i);\n");
```

Comment:
Thanks for the improvement. If you really want to improve performance of `toString` you should also look at the recursive call to `toString` in this line. It would be more efficient to pass the `StringBuilder` to the recursive call.